ext/web/06_streams.js

```diff
@@ -193,6 +207,20 @@
     return isFakeDetached in O;
   }
 
+  /**
+   * @param {ArrayBufferLike} O
+   * @returns {boolean}
+   */
+  function canTransferArrayBuffer(O) {
+    assert(typeof O === "object");
+    assert(O instanceof ArrayBuffer || O instanceof SharedArrayBuffer);
+    if (isDetachedBuffer(O)) {
+      return false;
+    }
+    // TODO(@crowlKats): 4. If SameValue(O.[[ArrayBufferDetachKey]], undefined) is false, return false.
```

Comment:
I believe this check is the same as [`v8::ArrayBuffer::is_detachable`](https://docs.rs/v8/0.33.0/v8/struct.ArrayBuffer.html#method.is_detachable). The use case for this seems to be to make it impossible to detach the buffer that backs a `WebAssembly.Memory` object other than through its `grow` method.

Reply:
Yes. I suggest we do this in a follow up, as it requires core bindings.